dev/conductor/ui/lib/widgets/progression.dart

```diff
@@ -25,17 +29,117 @@ class MainProgression extends StatefulWidget {
 }
 
 class MainProgressionState extends State<MainProgression> {
+  int _currentStep = 0;
+  // allow users to navigate back to already completed steps
```

Comment:
so what is the difference between `_currentStep` and `_navigateStep`?

Reply:
`_currentStep` keeps track of steps that have been completed, and `_navigateStep` keeps track of which step the user is viewing currently. For example. if a user completes 3/5 steps, but decides to click and visit the 1st step, `_currentStep` would be 2, and `_navigateStep` is 0.

Reply:
I noticed the naming could be improved. I changed the naming and added a dart doc to better explain the two variables.